server/src/main/java/org/spine3/server/stand/Stand.java

```diff
@@ -0,0 +1,663 @@
+     * @see #subscribe(Target)
+     */
+    public void activate(Subscription subscription, StandUpdateCallback callback) {
+        subscriptionRegistry.activate(subscription, callback);
+    }
+
+    /**
+     * Cancel the {@link Subscription}.
+     *
+     * <p>Typically invoked to cancel the previous {@link #activate(Subscription, StandUpdateCallback)} call.
+     * <p>After this method is called, the subscribers stop receiving the updates,
+     * related to the given {@code Subscription}.
+     *
+     * @param subscription a subscription to cancel.
+     */
+    public void cancel(Subscription subscription) {
+        subscriptionRegistry.removeSubscription(subscription);
+    }
+
+    /**
+     * Read all {@link Entity} types exposed for reading by this instance of {@code Stand}.
+     *
+     * <p>The result includes all values from {@link #getKnownAggregateTypes()} as well.
+     *
+     * @return the set of types as {@link TypeUrl} instances
+     */
+    @CheckReturnValue
+    public ImmutableSet<TypeUrl> getAvailableTypes() {
+        final ImmutableSet.Builder<TypeUrl> resultBuilder = ImmutableSet.builder();
+        final Set<TypeUrl> projectionTypes = typeToRepositoryMap.keySet();
+        resultBuilder.addAll(projectionTypes)
+                     .addAll(knownAggregateTypes);
+        final ImmutableSet<TypeUrl> result = resultBuilder.build();
+        return result;
+    }
+
+    /**
+     * Read all {@link org.spine3.server.aggregate.Aggregate} entity types exposed for reading
+     * by this instance of {@code Stand}.
+     *
+     * @return the set of types as {@link TypeUrl} instances
+     */
+    @CheckReturnValue
+    public ImmutableSet<TypeUrl> getKnownAggregateTypes() {
+        final ImmutableSet<TypeUrl> result = ImmutableSet.copyOf(knownAggregateTypes);
+        return result;
+    }
+
+    /**
+     * Read a particular set of items from the read-side of the application and feed the result into an instance
+     *
+     * <p>{@link Query} defines the query target and the expected detail level for response.
+     *
+     * <p>The query results are fed to an instance of {@link StreamObserver<QueryResponse>}.
+     *
+     * @param query            an instance of query
+     * @param responseObserver an observer to feed the query results to.
+     */
+    public void execute(Query query, StreamObserver<QueryResponse> responseObserver) {
+        final ImmutableCollection<Any> readResult = internalExecute(query);
+        final QueryResponse response = QueryResponse.newBuilder()
+                                                    .addAllMessages(readResult)
+                                                    .setResponse(Responses.ok())
+                                                    .build();
+        responseObserver.onNext(response);
+        responseObserver.onCompleted();
+    }
+
+    private ImmutableCollection<Any> internalExecute(Query query) {
```

Comment:
This method together with `feedEntitiesToBuilder()`, `fetchFromStandStorage()`, and `feedStateRecordsToBuilder()` can be parts of a helper class called `QueryRunner` or `QueryProcessor`. This way the `Stand` would be shorter and easier to understand.